colorcoder/Main.java

```diff
@@ -1,120 +1,38 @@
+import java.util.logging.Logger;
 
-    static class ColorPair {
-        private MajorColor majorColor;
-        private MinorColor minorColor;
-        
-        public ColorPair(MajorColor major, MinorColor minor)
-        {
-            majorColor = major;
-            minorColor = minor;
-        }
-        public MajorColor getMajor() {
-            return majorColor;
-        }
-        public MinorColor getMinor() {
-            return minorColor;
-        }
-        String ToString() {
-            String colorPairStr = MajorColorNames[majorColor.getIndex()];
-            colorPairStr += " ";
-            colorPairStr += MinorColorNames[minorColor.getIndex()];
-            return colorPairStr;
-        }
-    };
 
-    static ColorPair GetColorFromPairNumber(int pairNumber) {
-        int zeroBasedPairNumber = pairNumber - 1;
-        MajorColor majorColor = 
-            MajorColor.fromIndex(zeroBasedPairNumber / numberOfMinorColors);
-        MinorColor minorColor =
-            MinorColor.fromIndex(zeroBasedPairNumber % numberOfMinorColors);
-        return new ColorPair(majorColor, minorColor);
-    }
-    static int GetPairNumberFromColor(MajorColor major, MinorColor minor) {
-        return major.getIndex() * numberOfMinorColors + minor.getIndex() + 1;
-    }
+public class Main {
+		
+	private static final Logger LOGGER = Logger.getLogger(Main.class.getName());
+		
+	static ColorPair getColorFromPairNumber(int pairNumber) {
+		int zeroBasedPairNumber = pairNumber - 1;
+		MajorColor majorColor = MajorColor.fromIndex(zeroBasedPairNumber / ColorPair.nMajorColors);
+		MinorColor minorColor = MinorColor.fromIndex(zeroBasedPairNumber % ColorPair.nMinorColors);
+		return new ColorPair(majorColor, minorColor);
+	}
 
-    static void testNumberToPair(int pairNumber,
-        MajorColor expectedMajor,
-        MinorColor expectedMinor)
-    {
-        ColorPair colorPair = GetColorFromPairNumber(pairNumber);
-        System.out.println("Got pair " + colorPair.ToString());
-        assert(colorPair.getMajor() == expectedMajor);
-        assert(colorPair.getMinor() == expectedMinor);
-    }
+	static int getPairNumberFromColor(MajorColor major, MinorColor minor) {
+		return major.getIndex() * ColorPair.nMinorColors + minor.getIndex() + 1;
+	}
 
-    static void testPairToNumber(
-        MajorColor major,
-        MinorColor minor,
-        int expectedPairNumber)
-    {
-        int pairNumber = GetPairNumberFromColor(major, minor);
-        System.out.println("Got pair number " + pairNumber);
-        assert(pairNumber == expectedPairNumber);
-    }
+	
 
-    public static void main(String[] args) {
-        testNumberToPair(4, MajorColor.WHITE, MinorColor.BROWN);
-        testNumberToPair(5, MajorColor.WHITE, MinorColor.SLATE);
-    
-        testPairToNumber(MajorColor.BLACK, MinorColor.ORANGE, 12);
-        testPairToNumber(MajorColor.VIOLET, MinorColor.SLATE, 25);
-    }
+	static void printManual() {
```

Comment:
The name of this file doesn't give me a clue that I can find the manual-printing in here